espoon-voltti/evaka
# Keycloak saml

#### Summary

Keycloak SAML login PoC

Adds a new button to the employee UI to login as service provider.

(To try out with certificates: here https://github.com/espoon-voltti/evaka/pull/241 is a script to get and properly format those)

#### Dependencies
https://github.com/espoon-voltti/evaka-infra/pull/320

#### Testing instructions
On localhost:8080 (admin/admin):
1. Set up keycloak realm "evaka" (might be already set up)
2. Set up client "evaka" with the following settings:
<img width="1277" alt="Screenshot 2020-12-09 at 16 07 24" src="https://user-images.githubusercontent.com/158767/101640046-af62cf00-3a38-11eb-9b0e-66268ad13e3f.png">

AND in Fine grained SAML endpoint settings -> set the logout redirect setting to 'http://localhost:9093/employee/login'
AND in Client -> Mappers create a new javascript mapper called ID (all fields) with full source code as: "user.id"
3. create user
4. log in as the user (note that is has no roles, so add roles)
5. log in again

#### Checklist for pull request creator



#### Checklist for pull request reviewer (copy to review text box)


```
```
